pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/ForwardIndexHandler.java

```diff
@@ -660,12 +677,12 @@ private void writeDictEnabledForwardIndex(String column, ColumnMetadata existing
     }
   }
 
-  private void removeDictRelatedIndexes(String column, SegmentDirectory.Writer segmentWriter) {
+  static void removeDictRelatedIndexes(String column, SegmentDirectory.Writer segmentWriter) {
     // TODO: Move this logic as a static function in each index creator.
-    segmentWriter.removeIndex(column, ColumnIndexType.RANGE_INDEX);
+    DICTIONARY_RELATED_INDEX_TO_REMOVE.forEach((index) -> segmentWriter.removeIndex(column, index));
   }
 
-  private void updateMetadataProperties(File indexDir, Map<String, String> metadataProperties)
+  static void updateMetadataProperties(File indexDir, Map<String, String> metadataProperties)
```

Comment:
Hmm wondering if this will actually work in stateless / static fashion. Caller need not do anything after `properties.save()` ?

Reply:
Not sure I follow. `properties.save()` will save the properties into the metadata file. After this the next step is to cleanup indexes and stuff which is not related to this.